From 91f481d2046623389d3089269a5c997561e003d7 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Mon, 5 Feb 2018 16:50:08 +0800
Subject: [PATCH 3501/5242] MLK-17230-9: max9286: delete redundant 15fps
 support

commit  71b3d9c7c4bedd2ee65460b432747d86a7a362e7 from
https://source.codeaurora.org/external/imx/linux-imx.git

Delete redundant 15fps statements because max9286 does not
support.

Reviewed-by: Sandor.Yu <sandor.yu@nxp.com>
Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
(cherry picked from commit 88089e8a6dd5f0d06c7c61562f5a1942840f23e3)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/max9286.c |   10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/media/platform/imx8/max9286.c b/drivers/media/platform/imx8/max9286.c
index 73841a8..4c929cd 100644
--- a/drivers/media/platform/imx8/max9286.c
+++ b/drivers/media/platform/imx8/max9286.c
@@ -48,7 +48,6 @@ struct reg_value {
 };
 
 enum ov10635_frame_rate {
-	OV10635_15_FPS,
 	OV10635_30_FPS,
 };
 
@@ -2581,12 +2580,11 @@ static int max9286_s_parm(struct v4l2_subdev *sd, struct v4l2_streamparm *a)
 		tgt_fps = timeperframe->denominator /
 			  timeperframe->numerator;
 
-		if (tgt_fps == 15)
-			frame_rate = OV10635_15_FPS;
-		else if (tgt_fps == 30)
+		if (tgt_fps ==  30)
 			frame_rate = OV10635_30_FPS;
-		else {
-			pr_err(" The camera frame rate is not supported!\n");
+
+		if (frame_rate != OV10635_30_FPS) {
+			pr_err(" The camera %d frame rate is not supported!\n", frame_rate);
 			return -EINVAL;
 		}
 
-- 
1.7.9.5

